### PR TITLE
make cursor stay in the same position after buying

### PR DESCRIPTION
**Pull request by qndel**
effectively the same as https://github.com/diasurgical/devilutionX/pull/938 but less changes

### CODE REVIEW

File: Source/stores.cpp [hunk 1]
```diff
@@ -2543,6 +2547,18 @@ void S_ConfirmEnter()
 			break;
 		}
 		StartStore(stextshold);
+		if (forceSelection) {
+			stextsel = stextlhold;
+			stextsval = stextvhold; //making the selection stay after buying an item
+			if (stextsval >= 1) {
+				// this trick is the best way to use existing mechanics and make sure the selection is still valid
```
Comment (glebm) [inline Source/stores.cpp:2554]: Let's not use the trick and ensure in-range selection directly?
Then we won't need the `if (!forceSelection)` SFX disabling either

Comment (qndel) [reply Source/stores.cpp:2554]: are you saying I have any clue what's going on? :P

Comment (galaxyhaxz) [reply Source/stores.cpp:2554]: Seeing are you say what am I saying and see??

**Comment by AJenbo** [comment]
Thanks but this code is confusing enough already :D

Did this instead: https://github.com/diasurgical/devilutionX/commit/5d2af51d01a31865e7384ba1fb0d24a2b2c55d26

**Pull request closed without merging**